src/clr-angular/data/datagrid/datagrid-row.ts

```diff
@@ -214,12 +195,33 @@ export class ClrDatagridRow implements AfterContentInit {
         });
 
         // Used to set things up the first time but only after all the columns are ready.
-        this.subscription = this.hideableColumnService.columnListChange.subscribe((columnList) => {
+        this.subscriptions.push(this.hideableColumnService.columnListChange.subscribe((columnList) => {
             // Prevents cell updates when cols and cells array are not aligned - only seems to run on init / first time.
             if (columnList.length === this.dgCells.length) {
                 this.updateCellsForColumns(columnList);
             }
-        });
+        }));
+    }
+
+    ngAfterViewInit() {
+        this.subscriptions.push(this.displayMode.view.subscribe(viewChange => {
+            // Listen for view changes and move cells around depending on the current displayType
+            while (this.scrollableCells.detach()) {
+            }  // remove cells containers
+            while (this.calculatedCells.detach()) {
+            }  // remove cells containers
+            if (viewChange === DatagridDisplayMode.CALCULATE) {
+                this.dgCells.forEach(cell => {
```

Comment:
Serious performance question: does this run **before or after** we set the `display:none` on the display div in `datagrid.ts`?

It depends on who subscribes first I guess, but if for some reason this cell-moving happens before the `IE performance again.